### DISCUSSION TITLE
Migration after sunsetting

### DISCUSSION THREAD

**Discussion by arminvburren** [fief-dev/fief#343] [Q&A]
Hello,

So I guess I'm currently Fief Cloud with my Heroku apps, which has been very helpful. I'm seeing that there will be a sunsetting. How can i migrate all my users to another instance ?

Is there a way to host for free or do we have to use a paid provider necessarily ? Is there a way to host the server on heroku for a python App ?

Regards,

**Comment by frankie567** [comment]
Hi @arminvburren 👋 

I've added some instructions in the documentation to help you export your data from Fief Cloud and import it on your own instance: https://docs.fief.dev/migration/#fief-cloud

Unfortunately, it's nowadays quite rare to find a free hosting offer. It should be possible to deploy on Heroku, even though deploying a Docker image [looks like quite a challenge on this platform](https://devcenter.heroku.com/categories/deploying-with-docker). I estimate the cost around 26$/month (2 Basic Dynos + 1 Basic Postgres + 1 Mini Redis). Knowing that you can probably reuse the Postgres you probably have for your app.

[estimate-48dbe095585fbb0460151708b7e33c8450994de688e1a476f56733535c9fd233.pdf](https://github.com/fief-dev/fief/files/14479830/estimate-48dbe095585fbb0460151708b7e33c8450994de688e1a476f56733535c9fd233.pdf)


If you're interested, I might look into setting up a template (aka "Deploy to Heroku" button) for Heroku.

**Comment by arminvburren** [comment]
That could be great indeed. Is there not a way to just use the fief python code for authentification within the deployed app instead of just setting up a separate server ?

On another note, I'm surprised you said the reason for fief cloud sunsetting is the pricing model and it's a little bit bumming out. When I chose fief I really thought it was one of the easiest / simple lib to set up (and had the benefit of being free since my apps don't bring money in). That said now I'd be ready to pay something like 2 dollars / month or something.

Is there not a way to make that into a real business model ? I felt (and still feel) that if you did some marketing / promotion of the service a LOT of simple devs like me would rather chose such an option rather than all oauth provider on heroku who start at 15€ / month. Don't you want to make some promotion on stuff like hackernews or other ? That could quickly bring your business to breakeven maybe ?

Regards

**Reply by frankie567** [reply]
I'll look into that!

> . Is there not a way to just use the fief python code for authentification within the deployed app instead of just setting up a separate server ?

That's *theoretically* possible but quite hard in practice since your app would need to use FastAPI *and* follow the same Python and version libraires that Fief uses.

> On another note, I'm surprised you said the reason for fief cloud sunsetting is the pricing model and it's a little bit bumming out.

The main reason is that most people were actually not interested in the cloud offer. I answered dozens of questions related to self-hosting and how it was hard for them.

On Fief Cloud, we had only 3 really used instances (including you and a project of mine). And it was free. Can't imagine what it would have looked like if I charged for it, which was the initial plan. For the record, Fief Cloud costs me 50$ per month in hosting bills.

Maybe yes I could have done better regarding self-promotion and all. But the feedback I had from users was that the hosted offer was not the selling-point, and they preferred to host is themselves.

That said, I'm not closed to propose managed hosting offers, with one client = one server. But clearly, that would be more than 2$/month.

**Comment by arminvburren** [comment]
- That's theoretically possible but quite hard in practice since your app would need to use FastAPI and follow the same Python and version libraires that Fief uses.

Mm my websites are literally written in FastAPI, and I dont see why the rest of the libs should be a problem. That would be awesome to be able to do a tuto for this. This could also be great for other people whose API is in Fast API I guess ?

 - Fief Cloud costs me 50$ per month in hosting bill

Mm thanks I suppose :smile: ! Well I guess I don't get the details but honestly the Fief Cloud was a dream alternative compared to other stuff that were expensive and complicated. I'm not sure I understand the details but it seemed you could easily find 25 more people like me (if not 1000s) who would happily pay 2 euros per month for such as simple service.

Honesly have you tried HackedNews ? Or some WebDev forum ? What about a basic website template with basic auth free on github ? I literally thought of doing this a few months ago since I've myself built like 4 websites with fief since then

EDIT: out of curiosity who exactly are your power users ?
